packages/versioning/src/validate.ts

```diff
@@ -164,6 +172,95 @@ export function $onValidate(program: Program) {
   validateVersionedNamespaceUsage(program, namespaceDependencies);
 }
 
+function getAllVersions(p: Program, t: Type): Version[] | undefined {
+  const [namespace, _] = getVersions(p, t);
+  if (namespace === undefined) return undefined;
+
+  return getVersion(p, namespace)?.getVersions();
+}
+
+/**
+ * Ensures that properties whose type has changed with versioning are valid.
+ */
+function validateMultiTypeReference(program: Program, source: Type) {
+  const versionTypeMap = getVersionedTypeMap(program, source);
+  if (versionTypeMap === undefined) return;
+  for (const [version, type] of versionTypeMap!) {
```

Comment:
```suggestion
  for (const [version, type] of versionTypeMap) {
```
shouldn't need that as you already check for it above